### ISSUE TITLE
reorganize NumericalAlgorithms/Interpolation

### ISSUE THREAD

**Issue by kidder** [sxs-collaboration/spectre#1095]
This can be done after the current PRs are in, but this directory is getting cluttered.  I would propose adding two subdirectories `Actions` (as we currently do in other directories that have actions) and `Targets` (as I imagine there will be a few more of these).  Also I would shorten names of files and structs.  Files don't need Interpolation (or Target) in the name as they are in the appropriate dirs.
Structs don't need Interpolation in their name if they are in the `interp` namespace

**Comment by fmahebert** [comment]
Reminder from discussion in PR 1093: also need to clean up the tests, and the file `tests/Unit/NumericalAlgorithms/Interpolation/InterpolationTargetTestHelpers.hpp` should be moved along with the Target tests.

**Comment by markscheel** [comment]
I propose 3 directories: Actions, Targets, and Callbacks.

**Comment by nilsvu** [comment]
`Actions` makes sense, since those are general. Callbacks and Targets will be provided by the component that _uses_ the interpolation framework, so I think those should move there. Else, the interpolation framework will eventually depend on all modules that use it, whereas it should be the other way round.

**Comment by markscheel** [comment]
I agree about `Actions`.   Most of the `Actions` are internal things that an end-user will not see (e.g. Actions that communicate between InterpolationTargets and Interpolators), and are general.

The `Targets` are also Actions, but I think  of them differently than the other Actions because the user chooses one of them for each set of points that should be interpolated to. Three of the Targets are general (`KerrHorizon`, `LineSegment`, and `WedgeSectionTorus`) and don't depend on anything else (well `KerrHorizon` uses some Strahlkorper infrastructure), so I like putting these in `Targets`. Only one of them (`ApparentHorizon`) really has dependencies on something else, and I agree this one should be moved to somewhere in ApparentHorizons.

Same with `Callbacks`.  There are currently two of them.  One is general (ObserveTimeSeriesOnSurface) which I think should stay in `Callbacks`, and the other (FindApparentHorizon) should be moved to somewhere in ApparentHorizons.

**Comment by markscheel** [comment]
Earlier you (@nilsleiffischer) suggested renaming the `Targets` so that they are verbs, since they are Actions.  More descriptive names that are verbs would be `SendPointsOnWedgeSectionTorusToInterpolator`, `SendPointsOnTrialApparentHorizonToInterpolator`, etc.   I am torn on this one.  The longer names are more descriptive and tell you what these Targets actually do.  However, the longer names are more cumbersome.  In the Metavariables, do we want the user to type
```
using compute_target_points =  intrp::Actions::ApparentHorizon<AhA, ::Frame::Inertial>;
```
or

```
using compute_target_points =  
   intrp::Actions::SendPointsOnTrialApparentHorizonToInterpolator<AhA, ::Frame::Inertial>;
```

I'll leave that for discussion.

**Comment by markscheel** [comment]
Also, there is the question of moving stuff in the ApparentHorizons directory.  The only thing in the ApparentHorizons directory that is specific to only apparent horizons are FastFlow and the ComputeItems.   The other things (YlmSpherepack, Strahlkorper, and associated helpers) are lower-level infrastructure to describe 2-surfaces, and will be used for wave extraction and for Observers.  So possibly there can be two directories instead of a single ApparentHorizons directory that contains all of that.

Again, we probably want everyone to weigh in on this.

**Comment by markscheel** [comment]
Yet another thing about `Targets`.  This is something I've talked about with @fmahebert.  Currently all the Targets are Actions.  But it should be possible to replace these with a single Action, templated by a helper struct that is not an Action (the struct would have an initialize function and a compute_points function), and then these helper structs can be named `ApparentHorizon`, `LineSegment`, `WedgeSectionTorus`, etc.

**Comment by nilsvu** [comment]
About the `Targets`: what you just commented last is what I meant to suggest. The targets don’t have to duplicate the send-to-interpolator function call, but only do the point computing for a general action. This means they can be named to reflect that.

**Comment by nilsvu** [comment]
Noting something I just came across: `InterpolationTarget.hpp` is not currently imported anywhere, since the tests make their own mock version of it. Therefore, it has become quite outdated.

**Comment by nilsvu** [comment]
Most of this issue was resolved by #3552. I opened #3561 to track the remaining part. Please open additional issues if there's more.
